packages/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx

```diff
@@ -303,6 +306,7 @@ function GridColumnsManagement(props: GridColumnsManagementProps) {
               onClick={() => toggleAllColumns(!allHideableColumnsVisible)}
               name={apiRef.current.getLocaleText('columnsManagementShowHideAllText')}
               label={apiRef.current.getLocaleText('columnsManagementShowHideAllText')}
+              density="compact"
```

Comment:
Does it make sense to set it (and other styling like spacing etc) according to `rootProps.density`?

Reply:
Potentially, yes. I am thinking about how the other UI elements should respond to the density setting but it's going to require a bit of experimentation. Will create a follow up issue to review this across the board.